# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/sirkull/anomy-mailtools/clamav.patch
# ROCK Linux is Copyright (C) 1998 - 2004 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./contrib/check_for_virus.orig	2001-08-13 12:10:10.000000000 -0400
+++ ./contrib/check_for_virus	2003-08-25 10:17:26.000000000 -0400
@@ -49,6 +49,23 @@
 		logger "virus check for $1: ok"
 		echo "CLEAN : OK"	
 	fi
+# ClamAV (Clam AntiVirus)
+elif test -x /usr/bin/clamdscan; then
+	STATUS=
+	/usr/bin/clamdscan --quiet "$1"
+	RETURNCODE=$?
+	if test $RETURNCODE -eq 1; then
+		STATUS="virus found"
+		RET=3
+	fi
+	if test -n "$STATUS"; then
+		INFO=`/usr/bin/clamdscan --disable-summary --stdout "$1"|cut -f2 -d' '`
+		logger "virus check for $1: VIRUS FOUND!! - $INFO"
+		echo "VIRUS : $INFO"
+	else
+		logger "virus check for $1: ok"
+		echo "CLEAN : OK"
+	fi
 # AVP (Kaspersky Anti-Virus for Linux)
 elif test -x /usr/bin/kavscanner; then
 	STATUS=
